### PR TITLE
feat(enums): HTTP interaction bot flag

### PR DESCRIPTION
**Pull request by MaskDuck**
## Checklist

- [ ] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by ooliver1 — COMMENTED** [review]
Where's the documentation?

**Comment by ooliver1** [comment]
This also severely clashes with **your own** #911, since they both edit this same enum in a different way.

File: docs/api.rst [hunk 1]
```diff
@@ -1495,6 +1495,9 @@ of :class:`enum.Enum`.
     .. attribute:: discord_certified_moderator
 
         The user is a Discord Certified Moderator.
+    .. attribute:: http_interactions_bot
+
+        The user is a bot which receives interactions over HTTP.
```
Comment (DenverCoder1) [inline docs/api.rst:1500]: Should probably make the docs be more similar to what it says at https://github.com/discord/discord-api-docs/blob/8f435d11d78ea1c3db167364befc733df860bdcc/docs/resources/User.md?plain=1#L80

```suggestion
    .. attribute:: bot_http_interactions

        The user is a bot that uses only HTTP interactions and is shown in the online member list.
```

File: nextcord/enums.py [hunk 2]
```diff
@@ -472,6 +472,7 @@ class UserFlags(IntEnum):
     verified_bot = 1 << 16
     verified_bot_developer = 1 << 17
     discord_certified_moderator = 1 << 18
+    http_interactions_bot = 1 << 19
```
Comment (DenverCoder1) [inline nextcord/enums.py:475]: See above
```suggestion
    bot_http_interactions = 1 << 19
```

**Review by DenverCoder1 — APPROVED** [review]

**Review by EmreTech — APPROVED** [review]

**Pull request merged**